nixops/state/state_helper.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+
+import json
+import collections
+import nixops.util
+
+
+def _subclasses(cls):
+    sub = cls.__subclasses__()
+    return [cls] if not sub else [g for s in sub for g in _subclasses(s)]
+
+def _create_resource_state(depl, type, name, id):
+    """Create a resource state object  of the desired type."""
```

Comment:
I don't really like this kind of dynamic type programming, but perhaps others don't mind it. Some kind of reason as to why it *has* to work like this would be required if it was my code base (which it isn't).

Reply:
@coretemp I wasn't the original author of this method, I just refactored https://github.com/NixOS/nixops/pull/969/commits/b30dbfecfae5cf71b48184ae0b7b76984f8dcdc2#diff-9057d989ccc91407c2e92cca96a185caL359

original author is @moretea 

anyway, I think it's hard to banish `dynamic type programming` since looking for subclasses of `nixops.resources.ResourceDefinition` is conventional thing